httpx/instrument.go

```diff
@@ -0,0 +1,54 @@
+package httpx
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/socialpoint-labs/bsk/metrics"
+)
+
+// InstrumentDecorator returns an adapter that instrument requests with some metrics:
+// - http.request_duration: requests duration
+// - http.requests: number of requests
+//
+// Metrics are tagged with the HTTP method, requests path, response status code and response status class.
+func InstrumentDecorator(met metrics.Metrics, t ...metrics.Tag) Decorator {
```

Comment:
There is already a function: InstrumentRequestDurationMetric

Reply:
Right, but that's for the httpx client.